formats/datatables/resources/ext.srf.formats.datatables.js

```diff
@@ -558,6 +558,14 @@
 				});
 			}
 
+			// Replace -1 in lengthMenu with 'all' label
+			var showAll = options.lengthMenu.indexOf( -1 );
+			if ( showAll !== -1 ) {
+				var labels = options.lengthMenu.slice();
```

Comment:
Why the `slice()`?

Reply:
Need to clone the array so that the original `options.lengthMenu` wouldn't be altered

Reply:
> so that the original options.lengthMenu wouldn't be altered.

Why?

Reply:
On Line 566, we set two arrays to the `lengthMenu`. The first array is for the numerical values (`-1` represents `all`, which is the original values), the second array is for the label (`all` represents `-1`).